lib/Parse/ParseExpr.cpp

```diff
@@ -2959,38 +2959,57 @@ ExprResult Parser::ParseBoundsExpression() {
 }
 
 bool Parser::ParseRelativeBoundsClause(ExprResult &Expr) {
```

Comment:
I'd recommend renaming this method to avoid confusion.  It could be called ParseRelativeBoundsClauseForDecl (or something like that).